app/src/test/java/org/simple/clinic/summary/PatientSummaryScreenControllerTest.kt

```diff
@@ -576,6 +598,166 @@ class PatientSummaryScreenControllerTest {
+    // given
+    val screenCreatedTimestamp = Instant.parse("2018-01-01T00:00:00Z")
+    whenever(bpRepository.bloodPressureCountImmediate(patientUuid)) doReturn 1
+    whenever(patientRepository.hasPatientDataChangedSince(patientUuid, screenCreatedTimestamp)) doReturn true
+
+    // when
+    setupController(
+        openIntention = ViewExistingPatient,
+        screenCreatedTimestamp = screenCreatedTimestamp
+    )
+    startMobiusLoop()
+    uiEvents.onNext(PatientSummaryBackClicked(patientUuid, screenCreatedTimestamp))
+
+    verify(ui).showScheduleAppointmentSheet(patientUuid, BACK_CLICK)
+    verifyNoMoreInteractions(ui)
+    clearInvocations(ui)
+
+    uiEvents.onNext(ScheduleAppointmentSheetClosed(BACK_CLICK))
+
+    // then
+    verify(ui).goToPreviousScreen()
+    verifyNoMoreInteractions(ui)
+  }
+
+  @Test
+  fun `when schedule appointment sheet is closed after clicking back after linking an ID with existing patient, go to home screen`() {
+    // given
+    val screenCreatedTimestamp = Instant.parse("2018-01-01T00:00:00Z")
+    val identifier = PatientMocker.bpPassportIdentifier()
+    whenever(bpRepository.bloodPressureCountImmediate(patientUuid)) doReturn 1
+    whenever(patientRepository.hasPatientDataChangedSince(patientUuid, screenCreatedTimestamp)) doReturn true
+
+    // when
+    setupController(
+        openIntention = LinkIdWithPatient(identifier),
+        screenCreatedTimestamp = screenCreatedTimestamp
+    )
+    startMobiusLoop()
+    uiEvents.onNext(PatientSummaryBackClicked(patientUuid, screenCreatedTimestamp))
+
+    verify(ui).showLinkIdWithPatientView(patientUuid, identifier)
+    verify(ui).showScheduleAppointmentSheet(patientUuid, BACK_CLICK)
+    verifyNoMoreInteractions(ui)
+    clearInvocations(ui)
+
+    uiEvents.onNext(ScheduleAppointmentSheetClosed(BACK_CLICK))
+
+    // then
+    verify(ui).goToHomeScreen()
+    verifyNoMoreInteractions(ui)
+  }
+
+  @Test
+  fun `when schedule appointment sheet is closed after clicking save from a new patient, go to home screen`() {
+    // given
+    val screenCreatedTimestamp = Instant.parse("2018-01-01T00:00:00Z")
+    whenever(bpRepository.bloodPressureCountImmediate(patientUuid)) doReturn 1
+    whenever(patientRepository.hasPatientDataChangedSince(patientUuid, screenCreatedTimestamp)) doReturn true
+
+    // when
+    setupController(
+        openIntention = ViewNewPatient,
+        screenCreatedTimestamp = screenCreatedTimestamp
+    )
+    startMobiusLoop()
+    uiEvents.onNext(PatientSummaryDoneClicked(patientUuid))
+
+    verify(ui).showScheduleAppointmentSheet(patientUuid, DONE_CLICK)
+    verifyNoMoreInteractions(ui)
+    clearInvocations(ui)
+
+    uiEvents.onNext(ScheduleAppointmentSheetClosed(DONE_CLICK))
+
+    // then
+    verify(ui).goToHomeScreen()
+    verifyNoMoreInteractions(ui)
+  }
+
+  @Test
+  fun `when schedule appointment sheet is closed after clicking save from an existing patient, go to previous screen`() {
```

Comment:
Rename this test to `when schedule appointment sheet is closed after clicking save from an existing patient, go to home screen`?

Reply:
Whoops, good catch! Will fix.